### ISSUE TITLE
Gelocation Plugin needs to keep DC coverage and 'location' table in sync

### ISSUE THREAD

**Issue by kevinpdavies** [ozmeka/ozmeka#20]
The geolocation plugin stores a point only location in a separate table. The main advantage of this approach is the ability to search by location efficiently, however, location data is not stored within the DC metadata. If a record is moved or harvested, this information is lost.

Ideally the geolocation plugin would keep DC coverage in sync with the DB table. This includes:
- adding records to the location table for existing records (based on DC:coverage) on plugin install
- updating DC:coverage when a user adds a new item or edits the location of an existing item

Harvesting is still an issue though. The harvesting plugin would need to update the location table (if the plugin is installed) with the DC:coverage during harvest.

This is the simplest approach to improve spatial attribute handling in Omeka and does not address more complex issues of polygons, extended metadata etc.

**Comment by ptsefton** [comment]
Can we do this sync using a hook? That is after an item is updated by
anything, check if it has a location and update the location table? That
means plugins don't need to be aware of each other.

On Thu, Apr 16, 2015 at 4:00 PM, Kevin Davies notifications@github.com
wrote:

> The geolocation plugin stores a point only location in a separate table.
> The main advantage of this approach is the ability to search by location
> efficiently, however, location data is not stored within the DC metadata.
> If a record is moved or harvested, this information is lost.
> 
> Ideally the geolocation plugin would keep DC coverage in sync with the DB
> table. This includes:
> - adding records to the location table for existing records (based on
>   DC:coverage) on plugin install
> - updating DC:coverage when a user adds a new item or edits the
>   location of an existing item
> 
> Harvesting is still an issue though. The harvesting plugin would need to
> update the location table (if the plugin is installed) with the DC:coverage
> during harvest.
> 
> This is the simplest approach to improve spatial attribute handling in
> Omeka and does not address more complex issues of polygons, extended
> metadata etc.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ozmeka/ozmeka/issues/20.

## 

Peter Sefton +61410326955 pt@ptsefton.com http://ptsefton.com
Gmail, Twitter & Skype name: ptsefton

**Comment by kevinpdavies** [comment]
Yes ideally the plugin would encapsulate the location table management. I'm still learning how the Omeka plugins work so will take any advice on how best to do this.

**Comment by kevinpdavies** [comment]
The omeka geolocation plugin now updates the location table with latitude/longitude during a harvest from geonode. This is using basic DCSV syntax for coverage. Now I know how this works I will update to support extended spatial elements for dc:coverage.
